components/uniset/src/uniset.rs

```diff
@@ -22,10 +28,26 @@ pub struct UnicodeSet {
     // Allows for traits of fixed size arrays
 
     // Implements an [inversion list.](https://en.wikipedia.org/wiki/Inversion_list)
+    #[serde(deserialize_with = "deserialize_inv_list")]
     inv_list: Vec<u32>,
     size: usize,
```

Comment:
Issue: `size` is another invariant that you need to check.

I think I'd prefer if we *only* serialized the inversion list, and computed `size` on the fly since we need to walk the inversion list anyway for validation purposes.